### ISSUE TITLE
PaddleClas训练出来的ResNet50模型，推理精度比pytorch版本的低很多

### ISSUE THREAD

**Issue by ChungTak** [PaddlePaddle/PaddleClas#1773]
1. PaddleClas release/2.3和PaddlePaddle 2.2
 3. 训练环境信息：
  a. 具体操作系统 Ubuntu 20.04
  b. Python版本号 Python3.7
  c. CUDA10.2 cudnn7
  d. Docker镜像 registry.baidubce.com/paddlepaddle/paddle:2.2.2-gpu-cuda10.2-cudnn7
 4. 详细的错误信息及相关log
     1) PaddleClas训练ResNet50自定义数据集12个分类，epochs 200.
     训练结果精度正常
`[Eval][Epoch 200][Avg]CELoss: 1.74249, loss: 1.74249, top1: 0.93325 `
 `[Eval][Epoch 200][best metric: 0.9370988449787144] `

  
     2) 但是使用 python3 tools/infer.py 推理 分数只有 0.19 -0.2 之间，确认过测试的图片没问题，使用pytorch训练出来的ResNet50分数能到0.9
     3) 推理命令是

```
python3 tools/infer.py \
     -c ./ppcls/configs/custom/ResNet50.yaml \
     -o Infer.infer_imgs=./test-images/ \
    -o Global.pretrained_model=./output/ResNet50/best_model 
```
      

附件：训练配置 [ResNet50.zip](https://github.com/PaddlePaddle/PaddleClas/files/8310780/ResNet50.zip)

**Comment by weisy11** [comment]
训练有加载pretrain吗？
训练时加上 -o Arch.pretrained=True 试试看

**Comment by ChungTak** [comment]
@weisy11 感谢回复，有加 -o Arch.pretrained=True 训练出来[best metric: 0.9370988449787144] 很高，就是推理分数很低~~很奇怪。
补上我的训练参数
```
python3 tools/train.py \
    -c ./ppcls/configs/custom/ResNet50.yaml  \
    -o Arch.pretrained=True  \
    -o Global.device=gpu
``

**Comment by weisy11** [comment]
如果训练精度跟Pytorch情况差不多的话，推理一般也不会差太多，这个看起来更像bug。可以先排查一些基础问题：
1.预测的时候有没有选错模型文件？
2.看看标签映射有没有写错，有没有跳过背景类
3.分析一下预测的标签分布和badcase的情况，看下有没有异常的情况

**Comment by cuicheng01** [comment]
可以导出模型看下类别数是否一致，以及，确认下PaddleClas的版本是否为最新。

**Comment by ChungTak** [comment]
> 

1.预测模型没有错，我只训练了一个output下只有一个目录
2. 跳过背景类是指？我直接从0开始标注，是要从1开始标注么？
3. 标签分布和badcase  这些不太懂

**Comment by ChungTak** [comment]
> 如果训练精度跟Pytorch情况差不多的话，推理一般也不会差太多，这个看起来更像bug。可以先排查一些基础问题： 1.预测的时候有没有选错模型文件？ 2.看看标签映射有没有写错，有没有跳过背景类 3.分析一下预测的标签分布和badcase的情况，看下有没有异常的情况

1.预测模型没有错，我只训练了一个output下只有一个目录
2. 跳过背景类是指？我直接从0开始标注，是要从1开始标注么？
3. 标签分布和badcase 这些不太懂

**Comment by weisy11** [comment]
> > 如果训练精度跟Pytorch情况差不多的话，推理一般也不会差太多，这个看起来更像bug。可以先排查一些基础问题： 1.预测的时候有没有选错模型文件？ 2.看看标签映射有没有写错，有没有跳过背景类 3.分析一下预测的标签分布和badcase的情况，看下有没有异常的情况
> 
> 1.预测模型没有错，我只训练了一个output下只有一个目录 2. 跳过背景类是指？我直接从0开始标注，是要从1开始标注么？ 3. 标签分布和badcase 这些不太懂

是的，要从1开始标注，预测的时候也要从1开始预测，0要留出来

**Comment by chiliuliu** [comment]
> -o Arch.pretrained=True
你好，我看最新版本的imagenet1k_label_list.txt，里面是从0开始的。请问这是修改了吗？
